docs/src/development/proposals/MEP4/README.md

```diff
@@ -1,97 +1,978 @@
+}
+
+e = {"permission": permissions["metal.v2.image.get"]} {
+  some id
+  input.request.method = "GET"
+  input.request.path = ["v2", "image", id]
+  glob.match(input.backend.resources, [","], id)
+}
+
+e = {"permission": permissions["metal.v2.image.list"]} {
+  input.request.method = "GET"
+  input.request.path = ["v2", "image"]
+}
+
+e = {"permission": permissions["metal.v2.image.create"]} {
+  some id
+  input.request.method = "PUT"
+  input.request.path = ["v2", "image", id]
+  input.request.project != ""
+  glob.match(input.backend.resources, [","], id)
+}
+
+e = {"permission": permissions["metal.v2.image.update"]} {
+  some id
+  input.request.method = "POST"
+  input.request.path = ["v2", "image", id]
+  input.request.project != ""
+  glob.match(input.backend.resources, [","], id)
+}
+
+e = {"permission": permissions["metal.v2.image.delete"]} {
+  some id
+  input.request.method = "DELETE"
+  input.request.path = ["v2", "image", id]
+  input.request.project != ""
+  glob.match(input.backend.resources, [","], id)
+}
+
+e = {"permission": permissions["metal.v2.admin.image.create"]} {
+  some id
+  input.request.method = "PUT"
+  input.request.path = ["v2", "image", id]
+  input.request.project == ""
+  glob.match(input.backend.resources, [","], id)
+}
+
+e = {"permission": permissions["metal.v2.admin.image.update"]} {
+  some id
+  input.request.method = "POST"
+  input.request.path = ["v2", "image", id]
+  input.request.project == ""
+  glob.match(input.backend.resources, [","], id)
+}
+
+e = {"permission": permissions["metal.v2.admin.image.delete"]} {
+  some id
+  input.request.method = "DELETE"
+  input.request.path = ["v2", "image", id]
+  input.request.project == ""
+  glob.match(input.backend.resources, [","], id)
+}
+```
+
+OPA input:
+
+```json
+{
+  "token": {
+    "user": "gerrit",
+    "tenant": "metal-stack"
+  },
+  "request": {
+    "method": "GET",                                                 // from request
+    "path": ["v2", "image", "00000000-0000-0000-0000-ac1f6b7befb2"], // split on request path
+    "project": "793bb6cd-8b46-479d-9209-0fedca428fe1"                // from request query param
+  },
+  "backend": {
+    // composed from the user's general permissions and the permissions of the project in the request's query param
+    "permissions": "{metal.v2.image.*}",
+    "resources": "{*}"
+  }
+}
+```
+
+OPA output:
+
+```json
+{
+    "decision": {
+        "allow": true
+    },
+    "e": {
+        "permission": "metal.v2.image.get"
+    },
+    "permissions": [
+        "metal.v2.admin.image.create",
+        "metal.v2.admin.image.delete",
+        "metal.v2.admin.image.update",
+        "metal.v2.image.create",
+        "metal.v2.image.delete",
+        "metal.v2.image.get",
+        "metal.v2.image.list",
+        "metal.v2.image.update"
+    ]
+}
+```
+
+### Endpoints
+
+This section defines the new endpoints for the API.
+
+❓ Should we make "list and search" a single endpoint?
```

Comment:
One command(i would vote for `list`) is easier to memorize;)

Reply:
I am also leaning towards a single `list` endpoint (with optional filter payload). Should safe quite a lot of lines in the end and also closer to what K8s is doing,